hw/tpm/tpm_tis_spi.c

```diff
@@ -0,0 +1,220 @@
+#include "qemu/osdep.h"
+#include "hw/qdev-properties.h"
+#include "migration/vmstate.h"
+#include "hw/acpi/tpm.h"
+#include "tpm_prop.h"
+#include "tpm_tis.h"
+#include "qom/object.h"
+#include "hw/ssi/ssi.h"
+
+typedef enum {
+    STATE_IDLE,
+    STATE_ADDRESS,
+    STATE_DATA,
+} CMDState;
+
+union TpmTisRWSizeByte {
+    uint8_t byte;
+    struct {
+        uint8_t data__expected_size: 6;
+        uint8_t resv: 1;
+        uint8_t rwflag: 1;
+    }
+}
+
+union TpmTisSpiHwAddr {
+    hwaddr addr;
+    uint8_t bytes[sizeof(hwaddr)];
+}
+
+union TpmTisSpiData {
+    uint32_t data;
+    uint8_t bytes[sizeof(data)];
+}
+
+struct TPMStateSpi {
+    /*< private >*/
+    SSIPeripheral parent_obj;
+
+    /*< public >*/
+    TPMState state; /* not a QOM object */
+    uint8_t spi_state;
+
+    union TpmTisRWSizeByte first_byte;
+    union TpmTisSpiHwAddr addr;
+    union TpmTisSpiData data;
+
+    uint8_t data_idx;
+    uint8_t addr_idx;
+};
+
+OBJECT_DECLARE_SIMPLE_TYPE(TPMStateSpi, TPM_TIS_SPI)
+
+static uint32_t tpm_tis_spi_transfer8(SSIPeripheral *ss, uint32_t tx)
+{
+    TPMStateSpi *tts = TPM_TIS_SPI(ss);
+    uint32_t r = 0;
```

Comment:
As TPM_TIS_SPI wait state is 0, and we will never insert wait state, so let by default output 1.
```suggestion
    uint32_t r = 1;
```